tests/BucketerTest.php

```diff
@@ -291,34 +288,35 @@ public function testBucketInvalidExperiment()
 
         $this->assertEquals(
             new Variation(),
-            $bucketer->bucket($this->config, new Experiment(), $this->testBucketingIdControl, $this->testUserId)
+            $bucketer->bucket($this->config, new Experiment(), "returnsBeforeBucketingIdIsEvaluated", $this->testUserId)
         );
     }
 
+    // check that a non null bucketing ID should bucket to a variation
+    // different than the one expected with the userId
     public function testBucketWithBucketingId()
     {
         $bucketer = new Bucketer($this->loggerMock);
         $experiment = $this->config->getExperimentFromKey('test_experiment');
 
-        // check that a non null bucketing ID should bucket to a variation
-        // different than the one expected with the userId
+        // using user id as bucketing id . Bucketing key: "testUserId7716830082" produces bucket num 3455
         $this->assertEquals(
             new Variation('7722370027', 'control'),
             $bucketer->bucket(
                 $this->config,
                 $experiment,
-                $this->testBucketingIdControl,
+                $this->testUserId,
                 $this->testUserId
             )
         );
 
-        // check that the null bucketing ID defaults to bucketing with the userId
```

Comment:
The bucketer does not know that bucketing ID = user ID if bucketing ID is null. It's role is simply to bucket based on a given bucketing id. It's the decision service::getVariation which holds the logic for  bucketing ID = user ID if bucketing ID is null. 

This test isn't reporting an issue, because a null is considered as an empty string in php and when concatenated with the experiment id, it produces some number by chance which gives us our expected variation. 
See Logs: 
Optimizely.DEBUG: Assigned bucket 3218 to user "testUserId" with bucketing ID "testBucketingIdControl!" and with bucketing key "testBucketingIdControl!7716830082".
[2017-10-09 11:17:22] Optimizely.INFO: User "testUserId" is in variation control of experiment test_experiment.
[2017-10-09 11:17:22] Optimizely.DEBUG: Assigned bucket 7428 to user "testUserId" with bucketing ID "" and with bucketing key "7716830082".
[2017-10-09 11:17:22] Optimizely.INFO: User "testUserId" is in variation variation of experiment test_experiment.